app/models/payment_intent.rb

```diff
@@ -36,8 +36,13 @@
 #  fk_rails_...  (recipient_id => contacts.id)
 #
 class PaymentIntent < ApplicationRecord
-  validates_presence_of :square_payment_id, :square_location_id
-  validates_uniqueness_of :square_payment_id, allow_nil: false
+  ORIGIN_TYPES = [
+    CUSTOM = 'custom'.freeze,
+    SQUARE = 'square'.freeze
+  ].freeze
+
+  validates_presence_of :square_payment_id, :square_location_id, if: -> { origin == SQUARE }
+  validates_uniqueness_of :square_payment_id, allow_nil: true
```

Comment:
Can we verify that this is true in prod before merging?

Reply:
@justintmckibben I'm not sure I follow - verifying all `square_payment_ids` are unique in the DB?

Reply:
@justintmckibben and @slants0 I went ahead and checked the Prod DB -- looks like everything is indeed unique :)

Reply:
Woot thanks!